Unable to build LLVM on CentOS 7

I'm trying to build LLVM on CentOS 7 with GCC 7.1.0. I'm seeing the following error message:
```
In file included from /blues/gpfs/home/ajstewart/spack/var/spack/stage/llvm-4.0.0-iwr5xsmpa2scpclnuhevp4od7pa3j7mq/llvm-4.0.0.src/tools/lldb/source/Utility/TaskPool.cpp:10:0:
/blues/gpfs/home/ajstewart/spack/var/spack/stage/llvm-4.0.0-iwr5xsmpa2scpclnuhevp4od7pa3j7mq/llvm-4.0.0.src/tools/lldb/include/lldb/Utility/TaskPool.h: In static member function ‘static std::future<typename std::result_of<_Functor(_ArgTypes ...)>::type> TaskPool::AddTask(F&&, Args&& ...)’:
/blues/gpfs/home/ajstewart/spack/var/spack/stage/llvm-4.0.0-iwr5xsmpa2scpclnuhevp4od7pa3j7mq/llvm-4.0.0.src/tools/lldb/include/lldb/Utility/TaskPool.h:122:12: error: ‘bind’ is not a member of ‘std’
       std::bind(std::forward<F>(f), std::forward<Args>(args)...));
            ^~~~
/blues/gpfs/home/ajstewart/spack/var/spack/stage/llvm-4.0.0-iwr5xsmpa2scpclnuhevp4od7pa3j7mq/llvm-4.0.0.src/tools/lldb/include/lldb/Utility/TaskPool.h:122:12: note: suggested alternative: ‘end’
       std::bind(std::forward<F>(f), std::forward<Args>(args)...));
            ^~~~
            end
```
Full build log [here](https://github.com/LLNL/spack/files/1203298/spack-build.txt).

--- a/var/spack/repos/builtin/packages/llvm/llvm_gcc7.patch
+++ b/var/spack/repos/builtin/packages/llvm/llvm_gcc7.patch
@@ -1,0 +1,10 @@
+--- a/tools/lldb/include/lldb/Utility/TaskPool.h	2016-09-06 16:57:50.000000000 -0400
++++ b/tools/lldb/include/lldb/Utility/TaskPool.h	2017-08-29 16:29:41.448584015 -0400
+@@ -28,6 +28,7 @@
+ 
+ #include <cassert>
+ #include <cstdint>
++#include <functional>
+ #include <future>
+ #include <list>
+ #include <queue>

--- a/var/spack/repos/builtin/packages/llvm/package.py
+++ b/var/spack/repos/builtin/packages/llvm/package.py
@@ -324,6 +324,10 @@ class Llvm(CMakePackage):
     conflicts('+clang_extra', when='~clang')
     conflicts('+lldb',        when='~clang')
 
+    # Github issue #4986
+    patch('llvm_gcc7.patch', when='@4.0.0+lldb ^gcc@7.0:')
+    patch('llvm_gcc7.patch', when='@4.0.1+lldb ^gcc@7.0:')
+
     def setup_environment(self, spack_env, run_env):
         spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
 

--- a/var/spack/repos/builtin/packages/llvm/package.py
+++ b/var/spack/repos/builtin/packages/llvm/package.py
@@ -325,7 +325,7 @@ class Llvm(CMakePackage):
     conflicts('+lldb',        when='~clang')
 
     # Github issue #4986
-    patch('llvm_gcc7.patch', when='@4.0.0:4.0.1+lldb ^gcc@7.0:')
+    patch('llvm_gcc7.patch', when='@4.0.0:4.0.1+lldb %gcc@7.0:')
 
     def setup_environment(self, spack_env, run_env):
         spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)

--- a/var/spack/repos/builtin/packages/llvm/package.py
+++ b/var/spack/repos/builtin/packages/llvm/package.py
@@ -325,8 +325,7 @@ class Llvm(CMakePackage):
     conflicts('+lldb',        when='~clang')
 
     # Github issue #4986
-    patch('llvm_gcc7.patch', when='@4.0.0+lldb ^gcc@7.0:')
-    patch('llvm_gcc7.patch', when='@4.0.1+lldb ^gcc@7.0:')
+    patch('llvm_gcc7.patch', when='@4.0.0:4.0.1+lldb ^gcc@7.0:')
 
     def setup_environment(self, spack_env, run_env):
         spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)

